jepsen/resources/corrupt-file.c

```diff
@@ -703,29 +707,29 @@ int corrupt(struct opts opts) {
 
 /* Deletes the snapshot directory recursively. */
 int clear_snapshots() {
-  size_t limit = sizeof(SNAPSHOT_DIR) + 10;
-  char *cmd = malloc(limit);
-  int written = snprintf(cmd, limit, "rm -rf '%s'", SNAPSHOT_DIR);
+  char cmd[sizeof(SNAPSHOT_DIR) + 10] = {0};
+  int written = snprintf(cmd, sizeof(cmd), "rm -rf '%s'", SNAPSHOT_DIR);
   if (written < 0) {
     fprintf(stderr, "error writing string: %d", written);
     return EXIT_INT;
   }
   int ret = system(cmd);
-  free(cmd);
   return ret;
 }
 
 /* Go go go! */
 int main (int argc, char **argv) {
   /* Parse args */
-  struct opts opts = default_opts();
+  opts_t opts = default_opts();
   error_t err = argp_parse (&argp, argc, argv, 0, 0, &opts);
   if (err != 0) {
     fprintf(stderr, "Error parsing args: %d\n", err);
+    free(opts.file);
```

Comment:
Oh good call, I missed this!

Reply:
`valgrind` is definitely your friend for finding missing `free()` calls.